ballsdex/core/utils/buttons.py

```diff
@@ -0,0 +1,39 @@
+import discord
+
+from discord.ui import View, Button
+
+class ConfirmChoiceView(View):
+    def __init__(self, interaction: discord.Interaction):
+        super().__init__(timeout=90)
+        self.value = None
+        self.interaction = interaction
+
+    async def interaction_check(self, interaction: discord.Interaction) -> bool:
+        if interaction.user != self.interaction.user:
+            await interaction.response.send_message(
+                "Only the original author can use this.", ephemeral=True
+            )
+            return False
+        if self.value is not None:
+            await interaction.response.send_message(
+                "You've already made a choice.", ephemeral=True
+            )
+            return False
+        return True
+
```

Comment:
Would suggest adding this to disable the buttons once the view timed out.
```suggestion

    async def on_timeout(self):
        for item in self.children:
            item.disabled = True
        await self.interaction.followup.edit_message("@original", view=self)

```